Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/legacy/RiskLevelResultMigrator.kt

```diff
@@ -67,9 +69,24 @@ class RiskLevelResultMigrator @Inject constructor(
         override val riskLevel: RiskLevel,
         override val calculatedAt: Instant
     ) : RiskLevelResult {
+        override val failureReason: RiskLevelResult.FailureReason? = null
         override val aggregatedRiskResult: AggregatedRiskResult? = null
         override val exposureWindows: List<ExposureWindow>? = null
         override val matchedKeyCount: Int = 0
         override val daysWithEncounters: Int = 0
     }
+
+    companion object {
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        internal fun mapRiskLevelConstant(value: Int): RiskLevel {
```

Comment:
why is this mapping necessary?

Reply:
We move to a new risklevel/failure data state, but on the first update of the user we still need to know what the old 1,2,3,4 risk level constants mean, to provide an initial value. We could omit the whole migration if we are okay that the users have their card switch from red to green, if they can't calculate the first risklevel after the update.

Reply:
Thanks!